src/main/kotlin/sipura/alg/GraphProperty.kt

```diff
@@ -114,4 +119,48 @@ object GraphProperty {
         }
         return true
     }
+
+    /**
+     * Calculates the diameter of the given graph [g] by performing a breadth-first-search from every vertex
+     * in the graph.
+     *
+     * Runtime: O(n * m)
+     * @return the diameter of the graph. 0 if the graph is empty. -1 if the graph is not connected.
+     */
```

Comment:
I'd add the technical definition of terms if possible, so users don't have to switch to wikipedia to look it up.
Also this resolves any ambiguity in case there are conflicting definitions (e.g. length of path by # vertices or # edges).